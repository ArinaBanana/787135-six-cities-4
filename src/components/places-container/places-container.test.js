import React from "react";
import renderer from "react-test-renderer";
import {MemoryRouter} from "react-router-dom";
import PlacesContainer from "./places-container.jsx";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";

const places = [
  {
    id: 7,
    title: `Foo bar`,
    price: 45,
    img: `path`,
    type: `apartment`,
    rating: 3.5,
    isPremium: false,
    isBookmark: true,
    coordinates: [52.3809553943508, 4.939309666406198],
    city: {
      name: `City`,
      coordinates: [52.38333, 4.9],
      zoom: 12
    }
  },
  {
    id: 9,
    title: `Foo`,
    price: 80,
    img: `path`,
    type: `room`,
    rating: 4,
    isPremium: true,
    isBookmark: true,
    coordinates: [52.3809553943508, 4.939309666406198],
    city: {
      name: `City`,
      coordinates: [52.38333, 4.9],
      zoom: 12
    }
  }
];

const markers = [
  {
    coordinates: [52.3909553943508, 4.85309666406198],
    iconUrl: `path`
  },
  {
    coordinates: [52.3809553943508, 4.939309666406198],
    iconUrl: `path`
  }
];

const mockStore = configureStore([]);

it(`Should render Places container`, () => {
  const store = mockStore({});

  const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <PlacesContainer
            places={places}
            currentLocation={`Amsterdam`}
            setActiveElement={() => {}}
            city={[52.38333, 4.9]}
            zoom={10}
            markers={markers}
            changeSortType={() => {}}
          />
        </MemoryRouter>
      </Provider>,
      {
        createNodeMock: () => {
          return document.createElement(`div`);
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
