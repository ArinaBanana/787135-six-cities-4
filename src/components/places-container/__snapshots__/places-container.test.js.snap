// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Places container 1`] = `
<div
  className="cities__places-container container"
>
  <section
    className="cities__places places"
  >
    <h2
      className="visually-hidden"
    >
      Places
    </h2>
    <b
      className="places__found"
    >
      2
       places to stay in 
      Amsterdam
    </b>
    <form
      action=""
      className="places__sorting"
      method=""
    >
      <span
        className="places__sorting-caption"
      >
        Sort by
      </span>
      <span
        className="places__sorting-type"
        onClick={[Function]}
        tabIndex="0"
      >
        Popular
        <svg
          className="places__sorting-arrow"
          height="4"
          width="7"
        >
          <use
            xlinkHref="#icon-arrow-select"
          />
        </svg>
      </span>
      
    </form>
    <div
      className="cities__places-list places__list tabs__content"
    >
      <article
        className="cities__place-card place-card"
        onMouseMove={[Function]}
      >
        
        <div
          className="place-card__image-wrapper cities__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="200"
              src="path"
              width="260"
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                45
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "70%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/place/7"
              onClick={[Function]}
            >
              Foo bar
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            apartment
          </p>
        </div>
      </article>
      <article
        className="cities__place-card place-card"
        onMouseMove={[Function]}
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="place-card__image-wrapper cities__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="200"
              src="path"
              width="260"
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                80
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "80%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="/place/9"
              onClick={[Function]}
            >
              Foo
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            room
          </p>
        </div>
      </article>
    </div>
  </section>
  <div
    className="cities__right-section"
  >
    <section
      className="cities__map map"
    >
      <div
        style={
          Object {
            "height": "635px",
          }
        }
      />
    </section>
  </div>
</div>
`;
